src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -392,6 +411,30 @@ export class ChatListItemRenderer extends Disposable implements ITreeRenderer<Ch
 		return !!isFullyRendered;
 	}
 
+	private renderTreeData(data: IChatResponseProgressFileTreeData, element: ChatTreeItem, disposables: DisposableStore, templateData: IChatListItemTemplate): { element: HTMLElement; dispose: () => void } {
+		const ref = this._treePool.get();
+		const tree = ref.object;
+
+		const didOpenListener = tree.onDidOpen((e) => {
+			if (e.element && !('children' in e.element)) {
+				this.commandService.executeCommand('vscode.open', e.element.uri);
+			}
+		});
+
+		tree.setInput(data).then(() => {
+			tree.layout();
```

Comment:
Sorry I don't follow, what should be checked for disposal? 

The data is always available upfront but I don't know of a sync workbench data tree (tried to match this to the explorer view impl as much as possible)

Reply:
The search view uses the `WorkbenchCompressibleObjectTree` which is sync. But if it has to be async then inside the `.then` the tree could have been reused by that point

Reply:
Got it, thanks!

Reply:
Are you planning on trying to switch to a sync tree? I think it would be worth it because working with trees is hard enough without having to deal with async issues too. Otherwise I'm ✅ but I wanted to check because automerge is on

Reply:
I tried switching to a sync tree and ran into a weird bug with the tree height being incorrect even after a `setChildren` and `layout` / `rerender`, I can send another PR when I figure that one out but TBH the async tree is proving to be less difficult right now

Reply:
I was expecting that we would have to work through something like that, if that happened then the async tree probably has a more subtle layout issue 😆 I can help troubleshoot it, just be glad your items are fixed height